client/src/routes/accounts.lazy.tsx

```diff
@@ -54,30 +62,108 @@ function AccountsPage() {
     signupData
   );
 
+  // Generalized change handler for both forms
   const handleChange = (name: string, value: string | boolean) => {
     if (intent === "register") {
+      // Update signup form data
       setSignupData((prevData) => ({
         ...prevData,
         [name]: value,
       }));
     } else {
+      // Update login form data
       setFormData((prevData) => ({
         ...prevData,
         [name]: value,
       }));
     }
   };
 
+  // Function to validate signup form data
+  const validateForm = (formData: typeof signupData) => {
```

Comment:
To prevent fields from being empty you can just add a check for it in this function. To make it even better you can add the required attribute to the textInput fields that need it